sql/src/main/java/io/crate/execution/dsl/projection/UpdateProjection.java

```diff
@@ -21,35 +21,65 @@
 
 package io.crate.execution.dsl.projection;
 
+import io.crate.expression.symbol.InputColumn;
+import io.crate.expression.symbol.SelectSymbol;
 import io.crate.expression.symbol.Symbol;
+import io.crate.expression.symbol.SymbolVisitors;
 import io.crate.expression.symbol.Symbols;
+import io.crate.metadata.RowGranularity;
+import io.crate.types.DataTypes;
+import org.elasticsearch.Version;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 
 import javax.annotation.Nullable;
 import java.io.IOException;
 import java.util.Arrays;
+import java.util.List;
 
-public class UpdateProjection extends DMLProjection {
+public class UpdateProjection extends Projection {
+
+    final Symbol uidSymbol;
 
     private Symbol[] assignments;
     // All values of this list are expected to be a FQN columnIdent.
     private String[] assignmentsColumns;
+
+    private Symbol[] outputs;
+
+    @Nullable
+    private Symbol[] returnValues;
+
     @Nullable
     private Long requiredVersion;
 
+    private boolean allOn4_1;
+
     public UpdateProjection(Symbol uidSymbol,
                             String[] assignmentsColumns,
                             Symbol[] assignments,
+                            Version version,
+                            Symbol[] outputs,
+                            @Nullable Symbol[] returnValues,
                             @Nullable Long requiredVersion) {
-        super(uidSymbol);
+        this.uidSymbol = uidSymbol;
         this.assignmentsColumns = assignmentsColumns;
         this.assignments = assignments;
+        this.allOn4_1 = version.onOrAfter(Version.V_4_1_0);
```

Comment:
We'll need to update all the BWC related stuff to version 4.2.0 before merging this to master

Reply:
👍